### PR TITLE
Catch error upon import of models more gracefully

### LINKED ISSUE #391: Broken plugin messes up watercycle.models
If a plugin's model breaks upon import, this will mess with the ewatercycle.models entry point. This should be avoided somehow (e.g. a try/except)

### PR DESCRIPTION
**Pull request by BSchilperoort**
The user will now be presented with a more clear error when a plugin is broken. I went for @Peter9192's solution from #391.

closes #391

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,8 +6,14 @@ Formatted as described on [https://keepachangelog.com](https://keepachangelog.co
 
 ## [Unreleased]
 
-## Changed
+### Added
+- Added the `input_var_names` property to the eWaterCycle model class, to accompany the existing `output_var_names` property ([#403](https://github.com/eWaterCycle/ewatercycle/pull/403)).
+- Added the `var_units` method to the eWaterCycle model class, to mirror PyMT ([#403](https://github.com/eWaterCycle/ewatercycle/pull/403)).
+- Added a note on the time units that eWaterCycle expectes models to provide to the documentation ([#403](https://github.com/eWaterCycle/ewatercycle/pull/403)).
```
Comment (Peter9192) [inline CHANGELOG.md]:
```suggestion
- Added a note on the time units that eWaterCycle expects models to provide to the documentation ([#403](https://github.com/eWaterCycle/ewatercycle/pull/403)).
```

**Review by sverhoeven — APPROVED** [review]
Works for me, tested with injected exception

```python
import ewatercycle.models
/home/stefanv/mambaforge/envs/ewatercycle/lib/python3.10/site-packages/esmvalcore/experimental/_warnings.py:13: UserWarning:
  Thank you for trying out the new ESMValCore API.
  Note that this API is experimental and may be subject to change.
  More info: https://github.com/ESMValGroup/ESMValCore/issues/498
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
File ~/git/eWaterCycle/ewatercycle/src/ewatercycle/models.py:34
     33 try:
---> 34     globals()[_model.name] = _model.load()
     35 except Exception as e:

File ~/mambaforge/envs/ewatercycle/lib/python3.10/site-packages/importlib_metadata/__init__.py:210, in EntryPoint.load(self)
    209 match = self.pattern.match(self.value)
--> 210 module = import_module(match.group('module'))
    211 attrs = filter(None, (match.group('attr') or '').split('.'))

File ~/mambaforge/envs/ewatercycle/lib/python3.10/importlib/__init__.py:126, in import_module(name, package)
    125         level += 1
--> 126 return _bootstrap._gcd_import(name[level:], package, level)

File <frozen importlib._bootstrap>:1050, in _gcd_import(name, package, level)

File <frozen importlib._bootstrap>:1027, in _find_and_load(name, import_)

File <frozen importlib._bootstrap>:1006, in _find_and_load_unlocked(name, import_)

File <frozen importlib._bootstrap>:688, in _load_unlocked(spec)

File <frozen importlib._bootstrap_external>:883, in exec_module(self, module)

File <frozen importlib._bootstrap>:241, in _call_with_frames_removed(f, *args, **kwds)

File ~/mambaforge/envs/ewatercycle/lib/python3.10/site-packages/ewatercycle_wflow/model.py:24
     22 logger = logging.getLogger(__name__)
---> 24 raise Exception('Kaput')
     27 class _SwapXY(BmiProxy):

Exception: Kaput

The above exception was the direct cause of the following exception:

ImportError                               Traceback (most recent call last)
Cell In[1], line 1
----> 1 import ewatercycle.models

File ~/git/eWaterCycle/ewatercycle/src/ewatercycle/models.py:44
     35     except Exception as e:
     36         msg = (
     37             "An error was raised when trying to load the plugin of model "
     38             f"'{_model.name}'.\n"
   (...)
     42             f"    pip uninstall {_model.value.split('.')[0]}"
     43         )
---> 44         raise ImportError(msg) from e
     47 class ModelSources(shared.Sources):
     48     """Dictionary filled with available models.
     49
     50         Examples:
   (...)
     74
     75     """

ImportError: An error was raised when trying to load the plugin of model 'Wflow'.
You can report the issue on the model's github repository, or on https://github.com/eWaterCycle/ewatercycle/issues
In the meantime, you can try uninstalling the plugin with:
    pip uninstall ewatercycle_wflow
```

**Comment by sverhoeven** [comment]
It says `ewatercycle_wflow` in error while package is called [ewatercycle-flow](https://github.com/eWaterCycle/ewatercycle-wflow/blob/main/pyproject.toml#L9). 

Can you add a `replace('_'', "-")` to get the package name? Or use importlib somehow to get the right pip package name.

**Comment by BSchilperoort** [comment]
> It says `ewatercycle_wflow` in error while package is called [ewatercycle-flow](https://github.com/eWaterCycle/ewatercycle-wflow/blob/main/pyproject.toml#L9).
> 
> Can you add a `replace('_'', "-")` to get the package name? Or use importlib somehow to get the right pip package name.

Good point! I forgot about that. Pip doesn't allow for underscores AFAIK, so a simple string replacement should do the job.

Although package names can deviate from the module name... For example `sklearn`. You do `import sklearn` but `pip install scikit-learn`. I'll see if I can get that info with importlib

**Comment by BSchilperoort** [comment]
image

New error. The code now tries to lookup the package name using `importlib.metadata.packages_distributions()`.

**Pull request merged**